.github/workflows/test.yaml

```diff
@@ -193,10 +193,71 @@ jobs:
         files: .coverage/coverage-*.cov
         verbose: true
 
-  spread:
+  spread-fundamental:
     uses: ./.github/workflows/spread-tests.yaml
     needs: [unit-tests, snap-builds]
     name: "spread ${{ matrix.group }}"
+    with:
+      # Github doesn't support passing sequences as parameters.
+      # Instead here we create a json array and pass it as a string.
+      # Then in the spread workflow it turns it into a sequence 
+      # using the fromJSON expression.
+      runs-on: '["self-hosted", "spread-enabled"]'
+      group: ${{ matrix.group }}
+      backend: ${{ matrix.backend }}
+      systems: ${{ matrix.systems }}
+      tasks: ${{ matrix.tasks }}
+      rules: ${{ matrix.rules }}
+      is-fundamental: true
+    strategy:
+      # FIXME: enable fail-fast mode once spread can cancel an executing job.
+      # Disable fail-fast mode as it doesn't function with spread. It seems
+      # that cancelling tasks requires short, interruptible actions and
+      # interrupting spread, notably, does not work today. As such disable
+      # fail-fast while we tackle that problem upstream.
+      fail-fast: false
+      matrix:
+        include:
+          - group: 'debian-req (fundamental)'
+            backend: google-distro-1
+            systems: 'debian-11-64'
+            tasks: 'tests/...'
+            rules: 'main'
+          - group: 'fedora (fundamental)'
+            backend: openstack
+            systems: 'fedora-40-64'
+            tasks: 'tests/...'
+            rules: 'main'
+          - group: 'ubuntu-focal (fundamental)'
+            backend: google
+            systems: 'ubuntu-20.04-64'
+            tasks: 'tests/...'
+            rules: 'main'
+          - group: 'ubuntu-noble (fundamental)'
+            backend: google
+            systems: 'ubuntu-24.04-64'
+            tasks: 'tests/...'
+            rules: 'main'
+          - group: 'ubuntu-no-lts (fundamental)'
+            backend: google
+            systems: 'ubuntu-24.10-64'
+            tasks: 'tests/...'
+            rules: 'main'
+          - group: 'ubuntu-core-20 (fundamental)'
```

Comment:
I think I proposed UC18 actually, do we want that? or both? given the work we are doing with FDE it does make sense to have 20 at least until things settle in that area

Reply:
I will add UC18 no problem!